pkg/cmd/agent/agent.go

```diff
@@ -188,13 +184,6 @@ func Execute(opts *AgentOpts) error {
 
 	configPrefix := env.GetConfigPathWithDefault(env.DefaultConfigMountPath)
 
-	// Initialize cluster exporting if it's enabled
-	if env.IsExportClusterCacheEnabled() {
-		cacheLocation := confManager.ConfigFileAt(path.Join(configPrefix, "cluster-cache.json"))
-		clusterExporter = clustercache.NewClusterExporter(clusterCache, cacheLocation, ClusterExportInterval)
-		clusterExporter.Run()
-	}
-
```

Comment:
Why remove this?

Reply:
I haven't figure out what to do with this part yet. I disabled it temporary to have something I can compile without changing too much to start with.

The change will modify serialization/deserialization format, potentially making this part unusable (depending on how it's used). 

I need more understanding how it's used before approaching it.

Reply:
Could this kind of feature removal be behind a flag at all?

Reply:
I'll put it back in the final version.

But it will likely fail to load any existing files.

Can you provide any extra information on how it's used?

Reply:
I think it's used by a few upstream dependencies that have a multicluster implementation. We can check if its necesary but we should probably issue a deprecation notice even if we can't find anything that uses it.

Reply:
I put it back